R/size.R

```diff
@@ -99,43 +99,49 @@ vec_size2 <- function(nx, ny) {
 #'
 #' vec_slice(mtcars, 1:3)
 vec_slice <- function(x, i) {
-  i <- get_slice_index(i, x)
+  .Call(vctrs_slice, x, maybe_missing(i))
+}
 
-  if (is.null(x)) {
-    NULL
-  } else if (is.data.frame(x)) {
+vec_slice_dispatch <- function(x, i) {
+  if (is.data.frame(x)) {
```

Comment:
Will you do data frames in the next PR? It seems like you already have most of the pieces in place.

Reply:
Yes, data frames and maybe `vec_restore()`